### DISCUSSION TITLE
Port is already in use (Even though it isn't) AFTER running `kill-port ... ...`

### DISCUSSION THREAD

**Discussion by amunrarara** [vercel/next.js#46830] [Help]
### Summary

This issue is now happening with _two_ of our dev team's machines (both Macbook Pro M1)

To start up our dev environment, we use specific ports; here's the line in package.json: ` "dev": "next dev -H 127.0.0.1 -p 3030",`

Attempting to run our client on :3030 and server on :4040 worked great for a few months, but suddenly the client began throwing `Port 3030 is already in use`. 

I ran `kill-port 3030`, which returns `Could not kill process on port 3030. No process running on port.` Ran `yarn dev` again, issue persists. Odd... 

I checked for a process on that port via `lsof -i -P | grep 3030`. No processes are running on that port. Ran `yarn dev` again, issue persists. Odd...

I went ahead and changed the port to 3300, everything works, for a few weeks. Then, the issue appears again, this time complaining about Port 3300 being already in use. Steps above repeated, issue persists. Very odd...

I moved to 3301. A good couple of weeks. But, the issue returns. AND, the issue appears **for the server application on port 4040 as well**. So now, both are failing to start, with the same error: `Port 3301 is already in use` and `Port 4040 is already in use`.

Adding to the oddity, _now_ Port 3030 and 3300 are open again, so I can switch back and start using them. 

What gives?

<img width="708" alt="Screen Shot 2023-03-06 at 3 54 46 PM" src="https://user-images.githubusercontent.com/21048358/223132473-9883cbba-17fa-4711-994f-daac6872b06c.png">

**Comment by icyJoseph** [comment]
Wow, talk about strange bugs... does this happens without turbopack too? or more recent Node.js versions?

## Edit

Updating Turbopack fixed it.

**Reply by amunrarara** [reply]
We have a core dependency that requires Node 14, so I haven't been able to test on a more recent version. I'll update Turbopack and see if the issue persists

**Reply by icyJoseph** [reply]
For the record, I had a similar issue last week, but it was port 5000, and it was some AirPlay service running on it. 

I couldn't kill it from the CLI, it would just restart over and over.

The only way to kill it was to deactivate the AirPlay feature.

However this doesn't look like your problem at all.

**Reply by amunrarara** [reply]
Hmm, that's interesting. Yeah, if AirPlay was purposefully keeping track of the ports I'm using for a NextJS dev environment, that'd be really suspicious :-) It seems to be something within NextJS that's causing the issue.

**Reply by amunrarara** [reply]
But hey! **Updating to Turbo 1.8.3** fixed the issue! How wild. Thanks for recommending it!

**Reply by icyJoseph** [reply]
Awesome! yeah, alpha technology has this pitfalls :) - don't forget to mark as answered 😅

**Reply by amunrarara** [reply]
I spoke too soon. This issue is _not_ resolved. Still occurring after update to 1.8.3

**Comment by amunrarara** [comment]
Unfortunately, updating to 1.8.3 did _not_ resolve the issue.
